packages/contracts-bedrock/scripts/deploy/ReadImplementationAddresses.s.sol

```diff
@@ -0,0 +1,166 @@
+        );
+        return _l1CrossDomainMessenger;
+    }
+
+    function l1ERC721Bridge() public view returns (address) {
+        require(_l1ERC721Bridge != address(0), "ReadImplementationAddressesOutput: l1ERC721Bridge not set");
+        return _l1ERC721Bridge;
+    }
+
+    function l1StandardBridge() public view returns (address) {
+        require(_l1StandardBridge != address(0), "ReadImplementationAddressesOutput: l1StandardBridge not set");
+        return _l1StandardBridge;
+    }
+
+    function optimismMintableERC20Factory() public view returns (address) {
+        require(
+            _optimismMintableERC20Factory != address(0),
+            "ReadImplementationAddressesOutput: optimismMintableERC20Factory not set"
+        );
+        return _optimismMintableERC20Factory;
+    }
+
+    function disputeGameFactory() public view returns (address) {
+        require(_disputeGameFactory != address(0), "ReadImplementationAddressesOutput: disputeGameFactory not set");
+        return _disputeGameFactory;
+    }
+
+    function mipsSingleton() public view returns (address) {
+        require(_mipsSingleton != address(0), "ReadImplementationAddressesOutput: mipsSingleton not set");
+        return _mipsSingleton;
+    }
+
+    function preimageOracleSingleton() public view returns (address) {
+        require(
+            _preimageOracleSingleton != address(0), "ReadImplementationAddressesOutput: preimageOracleSingleton not set"
+        );
+        return _preimageOracleSingleton;
+    }
+}
+
+contract ReadImplementationAddresses is Script {
+    function run(ReadImplementationAddressesInput _rii, ReadImplementationAddressesOutput _rio) public {
+        address[6] memory eip1967Proxies = [
+            address(_rii.delayedWETHPermissionedGameProxy()),
+            address(_rii.optimismPortalProxy()),
+            address(_rii.systemConfigProxy()),
+            address(_rii.l1ERC721BridgeProxy()),
+            address(_rii.optimismMintableERC20FactoryProxy()),
+            address(_rii.disputeGameFactoryProxy())
+        ];
+
+        bytes4[6] memory sels = [
+            _rio.delayedWETH.selector,
+            _rio.optimismPortal.selector,
+            _rio.systemConfig.selector,
+            _rio.l1ERC721Bridge.selector,
+            _rio.optimismMintableERC20Factory.selector,
+            _rio.disputeGameFactory.selector
+        ];
+
+        for (uint256 i = 0; i < eip1967Proxies.length; i++) {
+            IProxy proxy = IProxy(payable(eip1967Proxies[i]));
+            vm.prank(address(0));
+            _rio.set(sels[i], proxy.implementation());
+        }
+
+        vm.prank(address(0));
+        address l1SBImpl = IStaticL1ChugSplashProxy(address(_rii.l1StandardBridgeProxy())).getImplementation();
+        vm.prank(address(0));
+        _rio.set(_rio.l1StandardBridge.selector, l1SBImpl);
+
+        (address mipsLogic,) = _rii.opcmProxy().implementations(_rii.release(), "MIPS");
```

Comment:
I've added getters in the new OPCM redesign. https://github.com/ethereum-optimism/optimism/pull/12851

Reply:
Nice, that'll clean this up a lot.